libs/sysplugins/smarty_internal_templatecompilerbase.php

```diff
@@ -640,17 +640,18 @@ public function compilePHPFunctionCall($name, $parameter)
                         return $func_name . '(' . $parameter[ 0 ] . ')';
                     }
                 } else {
-					$first_param = array_shift($parameter);
-					$modifier = array_merge(array($name), $parameter);
-					// Now, compile the function call as a modifier
-					return $this->compileTag(
-						'private_modifier',
-						array(),
-						array(
-							'modifierlist' => array($modifier),
-							'value'        => $first_param
-						)
-					);
+
+					if (
+						!$this->smarty->loadPlugin('smarty_modifiercompiler_' . $name)
+						&& !isset($this->smarty->registered_plugins[Smarty::PLUGIN_MODIFIER][$name])
```

Comment:
Why it only checks `PLUGIN_MODIFIER` but not `PLUGIN_FUNCTION`?

Do I need to register `sprintf` (for example) as modifier and function at the same time?